src/strands/agent/agent.py

```diff
@@ -614,6 +648,8 @@ async def _run_loop(self, messages: Messages, invocation_state: dict[str, Any])
             for message in messages:
                 self._append_message(message)
 
+            invocation_state["output_schema"] = output_schema
```

Comment:
We should not be modifying the invocation_state; we should be passing this explicitly

The only reason we modify this elsewhere is for backwards compatibility purposes

Reply:
updated!